src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -531,17 +486,17 @@ private List<SimpleStatement> buildSortedSelectQueries(DBFilterBase.IDFilter add
     List<SimpleStatement> queries = new ArrayList<>(expressions.size());
     expressions.forEach(
         expression -> {
-          List<Object> collect = ExpressionBuilder.getExpressionValuesInOrder(expression);
-          final QueryOuterClass.Query query =
+          //          List<Object> collect =
+          // ExpressionBuilder.getExpressionValuesInOrder(expression);
+          final Query query =
               new QueryBuilder()
                   .select()
                   .column(columnsToAdd)
                   .from(commandContext.namespace(), commandContext.collection())
                   .where(expression)
                   .limit(maxSortReadLimit())
                   .build();
-          final SimpleStatement simpleStatement = SimpleStatement.newInstance(query.getCql());
-          queries.add(simpleStatement.setPositionalValues(collect));
+          queries.add(query.queryToStatement());
```

Comment:
Nice! Looks like new way of building is much more compact from caller side. :)
(based on removal of codelines in this class)